charts/tekton-apps/templates/triggers.yaml

```diff
@@ -0,0 +1,83 @@
+{{- range $project := .Values.apps }}
+{{- if $project.enabled }}
+{{- range $component := $project.components }}
+
+{{/* initialize `projectEnvironment` variable with either `$project.environment` if it exists or with `$.Values.environment` otherwise */}}
+{{ $projectEnvironment := ternary $project.environment $.Values.environment (hasKey $project "environment") -}}
+
+{{/* initialize `gitBranchPrefixes` variable with either `$component.eventlistener.gitWebhookBranches` list if it exists or `$.Values.gitBranchPrefixes` otherwise */}}
+{{ $gitBranchPrefixes := ternary $component.eventlistener.gitWebhookBranches $.Values.gitBranchPrefixes (hasKey $component.eventlistener "gitWebhookBranches") -}}
+
+{{- $data := dict "project"           $project
+                  "component"         $component
+                  "environment"       $projectEnvironment }}
+
+{{- $filter := ternary $component.eventlistener.filter (include "tekton-apps.eventlistener.filter" $gitBranchPrefixes)
+              (hasKey $component.eventlistener "filter") -}}
+---
+apiVersion: triggers.tekton.dev/v1beta1
+kind: Trigger
+metadata:
+  labels:
+    pipeline: {{ $component.pipeline }}
+    {{- $.Values.eventlistener.labelSelector | toYaml | nindent 4 }}
+  name: {{ include "tekton-apps.resourceName" (set (deepCopy $data) "suffix" "trigger") }}
```

Comment:
Can we use just `data` instead of its deep copy everywhere? Or looping works incorrectly in such a case?

Reply:
without deepCopy "set" modifies original dictionary - ex. adds "suffix" - it will be available not only in this invocation, but on following ones too.